api/agent/agent.go

```diff
@@ -334,23 +341,14 @@ func (a *agent) hotLauncher(ctx context.Context, callObj *call) {
 
 		curStats := callObj.slots.getStats()
 		isNeeded := isNewContainerNeeded(&curStats)
-		logger.WithFields(logrus.Fields{
-			"currentStats": curStats,
-			"isNeeded":     isNeeded,
-		}).Debug("Hot function launcher stats")
+		logger.WithFields(logrus.Fields{"currentStats": curStats, "isNeeded": isNeeded}).Debug("Hot function launcher stats")
 		if !isNeeded {
 			continue
 		}
-
-		ctxResource, cancelResource := context.WithCancel(context.Background())
-		logger.WithFields(logrus.Fields{
-			"currentStats": curStats,
-			"isNeeded":     isNeeded,
-		}).Info("Hot function launcher starting hot container")
+		logger.WithFields(logrus.Fields{"currentStats": curStats, "isNeeded": isNeeded}).Info("Hot function launcher starting hot container")
 
 		select {
-		case tok, isOpen := <-a.resources.GetResourceToken(ctxResource, callObj.Memory, uint64(callObj.CPUs), isAsync):
-			cancelResource()
+		case tok, isOpen := <-a.resources.GetResourceToken(ctx, callObj.Memory, uint64(callObj.CPUs), isAsync):
 			if isOpen {
```

Comment:
unrelated, but there's a bug here... user first-time submits an impossible request (huge mem), then we spin infinite-loop

Reply:
I'll fix that one up, with another PR.

Reply:
good catch. i can add (need to return after queueing a poopy slot, right ?)

Reply:
oh, nevermind, there's an else handling that case.. Sorry... If I can only read.. :-)

Reply:
i guess this assumes that we are getting memory from the route (i.e. for a given route, we'll never be able to satisfy the resource request locally since memory is static) -- if we have ability to change memory per call then this behavior is less savory.

Reply:
yes, if mem/cpu were different per call, then this logic will break...

Reply:
then slot key changes and results in entire new hot container.

Reply:
https://github.com/fnproject/fn/pull/716/commits/f058559e2a7f9561e84214bca5aae4d6f563893d